ocean_provider/utils/util.py

```diff
@@ -305,3 +309,12 @@ def check_asset_consumable(asset, consumer_address, logger, custom_url=None):
     logger.error(message, exc_info=1)
 
     return False, message
+
+
+def check_environment_exists(envs, id):
+    """Checks if enironment with id exists in environments list."""
+    if envs and isinstance(envs, list):
```

Comment:
is there a more Pythonic way to do this? I don't like the triple imbrication. Why are we iterating with a range-like for? Can we check with [env for env in envs if env["id"] == id] and take the first item?

I suggest:
```
if not envs or not isinstance(envs, list):
     return False

matching_envs = [env for env in envs if env["id"] == id]

return len(matching_envs) > 0
```